pkg/controllers/nodeclass/validation.go

```diff
@@ -40,6 +60,125 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
+		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("create launch template %w", err))
+	}
+
+	amis := nodeClass.Status.AMIs
+	if amis == nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list AMIs from NodeClass status, %w", err)
+	}
+
+	var instanceType ec2types.InstanceType
+	if len(nodeClass.Status.AMIs) > 0 {
+		for _, req := range nodeClass.Status.AMIs[0].Requirements {
+			if req.Key == "kubernetes.io/arch" && len(req.Values) > 0 {
+				switch req.Values[0] {
+				case "amd64":
+					instanceType = ec2types.InstanceTypeT3Micro
+				case "arm64":
+					instanceType = ec2types.InstanceTypeT4gMicro
+				}
+				break
+			}
+		}
+	}
+
+	runInstancesInput := &ec2.RunInstancesInput{
+		DryRun:       lo.ToPtr(true),
+		MaxCount:     aws.Int32(1),
+		MinCount:     aws.Int32(1),
+		InstanceType: instanceType,
+		MetadataOptions: &ec2types.InstanceMetadataOptionsRequest{
+			HttpTokens:              ec2types.HttpTokensStateRequired,
+			HttpEndpoint:            ec2types.InstanceMetadataEndpointStateEnabled,
+			HttpPutResponseHopLimit: aws.Int32(1),
+		},
+		TagSpecifications: []ec2types.TagSpecification{
+			{
+				ResourceType: ec2types.ResourceTypeInstance,
+				Tags:         utils.MergeTags(tags),
+			},
+			{
+				ResourceType: ec2types.ResourceTypeVolume,
+				Tags:         utils.MergeTags(tags),
+			},
+			{
+				ResourceType: ec2types.ResourceTypeNetworkInterface,
+				Tags:         utils.MergeTags(tags),
+			},
+		},
+		ImageId: lo.ToPtr(amis[0].ID),
+	}
+
+	if _, err = n.ec2api.RunInstances(ctx, runInstancesInput); awserrors.IgnoreDryRunError(err) != nil {
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "RunInstancesAuthCheckFailed", fmt.Sprintf("unauthorized operation %s", err))
+		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("run instances %w", err))
+	}
 	nodeClass.StatusConditions().SetTrue(v1.ConditionTypeValidationSucceeded)
 	return reconcile.Result{}, nil
 }
+
+func mockLaunchTemplateConfig() []ec2types.FleetLaunchTemplateConfigRequest {
+	return []ec2types.FleetLaunchTemplateConfigRequest{
+		{
+			LaunchTemplateSpecification: &ec2types.FleetLaunchTemplateSpecificationRequest{
+				LaunchTemplateName: aws.String("mock-lt-name"),
+				LaunchTemplateId:   aws.String("lt-1234567890abcdef0"),
+				Version:            aws.String("1"),
+			},
+			Overrides: []ec2types.FleetLaunchTemplateOverridesRequest{
+				{
+					InstanceType: ec2types.InstanceTypeT3Micro,
+					SubnetId:     aws.String("subnet-1234567890abcdef0"),
+				},
+				{
+					InstanceType: ec2types.InstanceTypeT3Small,
+					SubnetId:     aws.String("subnet-1234567890abcdef1"),
+				},
+			},
+		},
+	}
+}
+func mockOptions(nodeClaim karpv1.NodeClaim, nodeClass *v1.EC2NodeClass, tags map[string]string) *amifamily.LaunchTemplate {
+	return &amifamily.LaunchTemplate{
+		Options: &amifamily.Options{
+			Tags:            tags,
+			InstanceProfile: lo.FromPtr(nodeClass.Spec.InstanceProfile),
```

Comment:
Does it matter if this value is empty? Because it probably will be in most cases